lib/awful/rules.lua

```diff
@@ -342,10 +342,41 @@ end
 
 rules.add_rule_source("awful.rules", apply_awful_rules, {"awful.spawn"}, {})
 
+local blacklisted_snid = {}
+
+--- Make startup notification work for some clients like XTerm. This is ugly
+-- but works often enough to be useful.
+local function get_startup_id_fallback(c)
+    if not c.pid then return end
+
+    local f = io.open("/proc/"..c.pid.."/environ", "rb")
+
+    -- It will only work on Linux, that's already 99% of the userbase.
+    if not f then return end
+
+    local s = f:read("*all")
+
+    f:close()
+
+    local snid = s:match("STARTUP_ID=(.*)")
+
+    -- There is an extra \0 in the string
+    snid = snid and snid:sub(1, -2) or nil
+
+    -- They can be used only once. Otherwise it is an inherited environ or an
+    -- app with multiple windows.
+    snid = snid and (not blacklisted_snid[snid]) and snid or nil
```

Comment:
`blackliste_snid` is always an empty table, so this condition will always evaluate to `true`...?